plugins/languages/otter.nix

```diff
@@ -0,0 +1,58 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "otter";
+  originalName = "otter.nvim";
+  defaultPackage = pkgs.vimPlugins.otter-nvim;
+
+  maintainers = [ lib.maintainers.perchun ];
+
+  settingsOptions = {
+    lsp = {
+      hover = {
+        border = helpers.defaultNullOpts.mkListOf lib.types.str [
```

Comment:
I am unsure that it is possible to set type to `list of N elements`

Reply:
I think we'd need to make a custom type for that, using [`mkOptionType`](https://github.com/NixOS/nixpkgs/blob/69bee9866a4e2708b3153fdb61c1425e7857d6b8/lib/types.nix#L115-L195).

I don't think we go to this extent for other options like this though. IMO `listOf str` is fine.

Reply:
We'd need to do something similar to [`nonEmptyListOf`](https://github.com/NixOS/nixpkgs/blob/69bee9866a4e2708b3153fdb61c1425e7857d6b8/lib/types.nix#L570-L576), i.e. a custom `check` function that checks the length.

Again, I don't think this is needed. It is something that _could_ be tackled in another PR targeting any other options that'd also benefit.

Reply:
I am pretty sure it is not the only one option for border characters in NixVim, and so it would be useful tree-wide. So I will leave this for someone else in another PR.